src/vaadin-pro-grid-inline-editing-mixin.html

```diff
@@ -0,0 +1,206 @@
+    }
+
+    _getItemValue(column, model) {
+      return this.get(column.path, model.item);
+    }
+
+    _isEditColumn(column) {
+      return column.localName.toLowerCase() === 'vaadin-pro-grid-edit-column';
+    }
+
+    _enqueueStopEdit() {
+      // schedule stop on text-field blur
+      this._debouncerStopEdit = Polymer.Debouncer.debounce(
+        this._debouncerStopEdit,
+        Polymer.Async.timeOut.after(200),
+        this._stopEdit.bind(this, true));
+    }
+
+    _cancelStopEdit() {
+      // stop edit on outside click will always trigger notify resize.
+      // when tabbing within same row it might not be needed, so cancel
+      if (this._debouncerStopEdit) {
+        this._debouncerStopEdit.cancel();
+        delete this._debouncerStopEdit;
+      }
+    }
+
+    _enterEditFromEvent(e) {
+      const context = this.getEventContext(e);
+      const column = context.column;
+
+      if (context.item && this._isEditColumn(column)) {
+        if (e.target.localName === 'vaadin-text-field') {
+          return;
+        }
+
+        const path = e.composedPath();
+        const cell = path[path.indexOf(this.$.table) - 3];
+
+        if (!cell || cell.getAttribute('part').indexOf('details-cell') > -1) {
+          return;
+        }
+
+        if (this.__edited) {
+          this._cancelStopEdit();
+          const oldCell = this.__edited.cell;
+          const shouldResize = oldCell.parentNode !== cell.parentNode;
+          this._stopEdit(shouldResize);
+        }
+
+        this._startEdit(cell, column, true);
+      } else if (this.__edited) {
+        this._stopEdit(true);
+      }
+    }
+
+    _startEdit(cell, column, shouldResize) {
+      // cancel debouncer enqueued on blur
+      this._cancelStopEdit();
+
+      const model = this.__getRowModel(cell.parentElement);
+      this.__edited = {cell, column, model};
+      column._startCellEdit(cell, model);
+
+
+      if (shouldResize) {
+        this.notifyResize();
+      }
+
+      column._setCellEditValue(cell, this._getItemValue(column, model));
+    }
+
+    _stopEdit(shouldResize) {
+      if (!this.__edited) {
+        return;
+      }
+      const {cell, column, model} = this.__edited;
+
+      const value = column._getCellEditValue(cell);
+      if (value !== this._getItemValue(column, model)) {
+        this.dispatchEvent(new CustomEvent('item-property-changed', {
+          detail: {
+            index: model.index,
+            item: model.item,
+            path: column.path,
+            value: value
+          },
+          bubbles: true,
+          composed: true
+        }));
+      }
+
+      column._stopCellEdit(cell, model);
+
+      this.__edited = null;
+
+      if (shouldResize) {
+        this.notifyResize();
+      }
+    }
+
+    _switchEditCell(e) {
+      this._cancelStopEdit();
+
+      const columnTreeLevel = this._columnTree.length - 1;
+      const cols = this._columnTree[columnTreeLevel].filter(this._isEditColumn);
```

Comment:
I think you'll need to sort these by column `_order` which is the current effective order (drag-reordering doesn't actually reorder the column elements in the DOM). Also, some columns might be hidden so those should be filtered out. [Example](https://github.com/vaadin/vaadin-grid/blob/805e1158795f3563238d49c058dbf91c29b3e4ca/src/vaadin-grid-column-resizing-mixin.html#L44)